Core/clim-core/presentation-defs.lisp

```diff
@@ -1607,77 +1607,83 @@ protocol retrieving gestures from a provided string."))
+			 collect (subseq home-env-variable pos next-pos)
+			 until (null next-pos)))
+		(new-directory (cons
+				(first his-directory)
+				(append home (rest (rest his-directory))))))
+	     (make-pathname :host (pathname-host pathname)
+			    :device (pathname-device pathname)
+			    :name (pathname-name pathname)
+			    :version (pathname-version pathname)
+			    :type (pathname-type pathname)
+			    :directory new-directory)
+	     )))
+    ;; Slow but accurate
+    (let* ((raw-pathname (pathname string))
+	   (raw-directory (pathname-directory raw-pathname))
+	   (original-pathname (if (and (listp raw-directory)
+				       (eql (first raw-directory) :absolute)
+				       (eql (second raw-directory) :Home))
+				  (deal-with-home raw-pathname raw-directory)
+				  raw-pathname))
+	   (original-string (namestring original-pathname))
+	   ;; Complete logical pathnames as well as regular pathnames
+	   ;; strategy is to keep track of both original string provided and translated string
+	   ;; but to return pathname built from original components except for the name.
+	   (logical-pathname-p (typep original-pathname 'logical-pathname))
+	   (actual-pathname (if logical-pathname-p
+				(translate-logical-pathname original-pathname)
+				original-pathname))
+	   (merged-pathname (merge-pathnames actual-pathname default))
+	   completions)
+      (let ((search-pathname (make-pathname :host (pathname-host merged-pathname)
+					    :device (pathname-device merged-pathname)
+					    :directory (pathname-directory merged-pathname)
+					    :version :unspecific
+					    :type :wild
+					    :name :wild)))
+	(setq completions (directory search-pathname)))
+      ;; Now prune out all completions that don't start with the string
+      (let ((type (pathname-type actual-pathname)))
+	(when (null type)
+	  ;; If the user didn't supply a file type, don't burden him with all
+	  ;; sorts of version numbers right now.
+	  (let ((new-completions nil))
+	    (dolist (pathname completions)
+	      (cond
+		;; meaning this is actually a directory
+		((and (null (pathname-name pathname))
+		      (null (pathname-type pathname)))
+		 (pushnew (make-pathname :host (pathname-host original-pathname)
+					 :device (pathname-device original-pathname)
+					 :directory (butlast (pathname-directory pathname))
+					 :name (first (last (pathname-directory pathname)))
+					 :type nil)
+			  new-completions))
+		(t
+		 (pushnew (make-pathname :host (pathname-host original-pathname)
+					 :device (pathname-device original-pathname)
+					 :directory (pathname-directory original-pathname)
+					 :name (pathname-name pathname)
+					 :type (pathname-type original-pathname))
+			  new-completions))))
+	    (setq completions (nreverse new-completions))))
+	(complete-from-possibilities original-string completions '(#\space)
+						    :action action
+						    :name-key #'namestring
+						    :value-key #'identity)))))
```

Comment:
new method has a regression with completion (disregarding the home directory). example failure mode:

1. have only one file in the directory called `hello`
2. open listener and type ,Show File /path/to/hello/h
3. complete with tab

expected (and matching old behavior): completes to `hello`
observed (in the new code): completes to `H` (capital H)